tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md

```diff
@@ -0,0 +1,606 @@
+
+
+4. Navigate to the object page
+
+    To add further tests, you can navigate from the list report to the object page.
+
+    To navigate to the object page, click on one of the loaded items in the list report table. You can do this by using the table action `iPressRow()`. The parameter corresponds to the one used in the `iSelectRows()` function. You can pass a zero- based line index or a condition which defines the row to be clicked. See the coding fragment below showing the test function using a condition to navigate to the object page.
+
+    ```Javascript
+    ...
+      When.onTheMainPage.onTable().iPressRow({ "Customer": "Prinz (608)" });
+    });
+    ```
+
+    The next test chapter in the journey `(#2: Object Page: Check Object Page loads)` still contains a check for the list report as the first test. This currently leads to an error now, since the navigation to the object page is already done and the list report is not visible anymore. To fix this, change the page object for the list report `onTheMainPage` to the one for the object page `onTheDetailPage`.
+
+    ```Javascript
+    opaTest('#2: Object Page: Check Object Page loads', function(Given, When, Then) {
+        Then.onTheDetailPage.iSeeThisPage();
+    });
+    ```
+
+
+[VALIDATE_3]
+
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 6: ](Testing the object page)]
+
+The object page contains further UI objects you can test using OPA tests. There are header elements, sections, forms, tables, etc. You can also navigate down to sub-object pages and edit and save data - so testing a complete workflow of loading/creating, changing and saving of data is possible.
+
+!![ObjectPage_1](ObjectPage_1.png)
+
+1. Checking fields on the object page header
+
+    To confirm that the correct record is loaded in the object page, check the title using the test function `iCheckTitle()` from the header assertions. Additionally, you can check if the item loaded can be edited by testing the properties `visible` and `enabled` of the Edit button. Add the `iCheckEdit()` assertion to the existing checks. Since this is another header assertion you don´t need to repeat the `onHeader()` command.
+
+    ```Javascript
+    ...
+      Then.onTheDetailPage
+          .iSeeThisPage()
+          .and.onHeader().iCheckTitle("Business Trip for Christine, Pierre")
+          .and.iCheckEdit({ visible: true, enabled: true });
+    });
+    ```
+
+2. Check a field in a form of a section
+
+    You can use the form assertions to check fields within the sections of an object page. As implemented for most of the test functions, you can use both strings and objects as parameters for identifying the UI controls you want to check. When you provide a string, the test functions expect the label of a control to be passed as a value. If an object is provided as the parameter, the internal name of the control is expected. The test function then creates the stable ID used for the control for identification.
```

Comment:
"You can use the form assertions to check fields " - If the word "from" is needed here, please put it in back-tics.